### ISSUE TITLE
could not install on nodejs 3.x/iojs

### ISSUE THREAD

**Issue by patriziobruno** [LinusU/node-generate-rsa-keypair#1]
``` log
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair>if not defined npm_config_node_gyp (node "c:\Program Files\iojs\node_modules\npm\bin\node-gyp-in\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
Building the projects in this solution one at a time. To enable parallel build,
please add the "/m" switch.
  binding.cc
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\node_modules\nan\nan.h(326): error C2995: 'v8::Local<T> Nan::imp::NanEnsureHandleOrPersistent(const v8::Local<T> &)' : function template has already been defined [C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\build\generate-rsa-keypair.vcxproj]
          C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypai  r\node_modules\nan\nan.h(319) : see declaration of 'Nan::imp::NanEnsureHandleOrPersistent'
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\node_modules\nan\nan.h(346): error C2995: 'v8::Local<T> Nan::imp::NanEnsureLocal(const v8::Local<T> &)' : function template has already been defined [C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\build\generate-rsa-keypair.vcxproj]
          C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypai  r\node_modules\nan\nan.h(334) : see declaration of 'Nan::imp::NanEnsureLocal'
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\node_modules\nan\nan.h(757): error C3083: 'smalloc': the symbol to the left of a '::' must be a type [C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\build\generate-rsa-keypair.vcxproj]
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\node_modules\nan\nan.h(757): error C2039: 'FreeCallback' : is not a member of 'node' [C
:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\build\generate-rsa-keypair.vcxproj]
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\node_modules\nan\nan.h(757): error C2061: syntax error : identifier 'FreeCallback' [C:\
Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\build\generate-rsa-keypair.vcxproj]
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\node_modules\nan\nan.h(761): error C2065: 'callback' : undeclared identifier [C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\build\generate-rs
a-keypair.vcxproj]
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\node_modules\nan\nan.h(761): error C2065: 'hint' : undeclared identifier
[C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\build\generate-rsa-keypair.vcxproj]
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\node_modules\nan\nan.h(768): error C2665: 'node::Buffer::New' : none of the 4 overloads could convert all the argument types [C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\build\generate-rsa-eypair.vcxproj]
          C:\Users\eul0860\.node-gyp\3.1.0\include\node\node_buffer.h(43): could be 'v8::MaybeLocal<v8::Object> node::Buffer::New(v8::Isolate *,char *,size_  t)'
          C:\Users\eul0860\.node-gyp\3.1.0\include\node\node_buffer.h(31): or
       'v8::MaybeLocal<v8::Object> node::Buffer::New(v8::Isolate *,v8::Local<v8::String>,node::encoding)'
          while trying to match the argument list '(v8::Isolate *, const char *
  , uint32_t)'
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\node_mod
ules\nan\nan.h(772): error C2664: 'v8::Local<v8::Object>::Local(const v8::Local
<v8::Object> &)' : cannot convert argument 1 from 'v8::MaybeLocal<v8::Object>'
to 'const v8::Local<v8::Object> &' [C:\Users\eul0860\AppData\Roaming\npm\node_m
odules\generate-rsa-keypair\build\generate-rsa-keypair.vcxproj]
          Reason: cannot convert from 'v8::MaybeLocal<v8::Object>' to 'const v8
  ::Local<v8::Object>'
          No user-defined-conversion operator available that can perform this c
  onversion, or the operator cannot be called
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\node_mod
ules\nan\nan.h(779): error C2039: 'Use' : is not a member of 'node::Buffer' [C:
\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\build\gene
rate-rsa-keypair.vcxproj]
C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keypair\node_mod
ules\nan\nan.h(779): error C3861: 'Use': identifier not found [C:\Users\eul0860
\AppData\Roaming\npm\node_modules\generate-rsa-keypair\build\generate-rsa-keypa
ir.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\12.0\bin\msbuild.exe` fail
ed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (c:\Program Files\iojs\node_modules\np
m\node_modules\node-gyp\lib\build.js:269:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_proces
s.js:200:12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "c:\\Program Files\\iojs\\node.exe" "c:\\Program Files\\iojs\\n
ode_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\eul0860\AppData\Roaming\npm\node_modules\generate-rsa-keyp
air
gyp ERR! node -v v3.1.0
gyp ERR! node-gyp -v v2.0.2
gyp ERR! not ok
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "c:\\Program Files\\iojs\\node.exe" "c:\\Program Files\\iojs\\node
_modules\\npm\\bin\\npm-cli.js" "install" "-g" "generate-rsa-keypair"
npm ERR! node v3.1.0
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! generate-rsa-keypair@0.1.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
```

**Comment by LinusU** [comment]
Nan 1.x doesn't work with node 3.x. The best solution is to update to Nan 2.x which should be really easy

**Comment by patriziobruno** [comment]
@LinusU issue updated

**Comment by LinusU** [comment]
Thx :+1:

**Comment by LinusU** [comment]
Should be fixed by #3, released as `0.1.1` on npm.
